spec/wcn_scraper/prison_spec.rb

```diff
@@ -142,8 +142,86 @@
+          { name: 'HMP Bourne', town: 'Bourne End', lat: 50.5495271, lng: -3.9963275, type: 'Youth Custody' }
+        ]
+        expect(prisons.map(&:attrs)).to match_array(expected_prisons)
+      end
+    end
+    context 'given a string containing two prisons, one of which has text contained in the other, ordered ascending' do
+      subject(:prisons) { described_class.find_in_string('201706: Prison Officer - HMP Bourne & HMP Bournemouth') }
+
+
+      it 'returns an array of Prison instances' do
+        expect(prisons).to all(be_a(described_class))
+      end
+
+      it 'returns two results' do
+        expect(prisons.count).to eq(2)
+      end
+
+      it 'returns the correct prisons' do
+        stub_const('PRISONS',
+                   [
+                     { name: 'HMP Bourne', town: 'Bourne End', lat: 50.5495271, lng: -3.9963275, type: 'Youth Custody' },
+                     { name: 'HMP Bournemouth', town: 'Bournemouth', lat: 0.5495271, lng: 2.9963275, type: 'Youth Custody' }
+                   ])
+
+        expected_prisons = [
+          { name: 'HMP Bournemouth', town: 'Bournemouth', lat: 0.5495271, lng: 2.9963275, type: 'Youth Custody' },
+          { name: 'HMP Bourne', town: 'Bourne End', lat: 50.5495271, lng: -3.9963275, type: 'Youth Custody' }
+        ]
+        expect(prisons.map(&:attrs)).to match_array(expected_prisons)
+      end
+    end
+    context 'given a string containing two prisons, one of which has text contained in the other, ordered descending' do
+      subject(:prisons) { described_class.find_in_string('201706: Prison Officer - HMP Bourne & HMP Bournemouth') }
+
+      it 'returns an array of Prison instances' do
+        expect(prisons).to all(be_a(described_class))
+      end
+
+      it 'returns two results' do
+        expect(prisons.count).to eq(2)
+      end
+
+      it 'returns the correct prisons' do
+        stub_const('PRISONS',
+                   [
+                     { name: 'HMP Bournemouth', town: 'Bournemouth', lat: 0.5495271, lng: 2.9963275, type: 'Youth Custody' },
+                     { name: 'HMP Bourne', town: 'Bourne End', lat: 50.5495271, lng: -3.9963275, type: 'Youth Custody' }
+                   ])
+
+        expected_prisons = [
+          { name: 'HMP Bournemouth', town: 'Bournemouth', lat: 0.5495271, lng: 2.9963275, type: 'Youth Custody' },
+          { name: 'HMP Bourne', town: 'Bourne End', lat: 50.5495271, lng: -3.9963275, type: 'Youth Custody' }
+        ]
+        expect(prisons.map(&:attrs)).to match_array(expected_prisons)
+      end
+    end
+    context 'given a string containing two prisons, the substring site should only return once' do
+      subject(:prisons) { described_class.find_in_string('201706: Prison Officer - HMP Bourne & HMP Bournemouth') }
+
+      it 'returns an array of Prison instances' do
+        expect(prisons).to all(be_a(described_class))
+      end
+
+      it 'returns two results' do
```

Comment:
This should read `it 'returns one result'`

Reply:
Done